### PR TITLE
[dashboard] user visible rename team->organization

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Renames all user visible occurences of the term team to organization

## Related Issue(s)

Fixes #

## How to test

check the UI and verify that we are talking about organizations everywhere.

## Release Notes

```release-note
teams are now called organizations
```

## Documentation

TODO

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-payment
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by easyCZ** [comment]
What's the plan for coordinating with docs changes for teams, and billing?

**Comment by easyCZ** [comment]
As a follow-up, we should also update error messages returned by the APIs which mention Teams, as those are also user facing (these get shown when the request fails)

**Comment by svenefftinge** [comment]
> What's the plan for coordinating with docs changes for teams, and billing?

I'll put this on hold and make sure the rename on docs/website happens in parallel. 
Just wanted to get some sort of signal first about the approach to only change the user-facing bit. 

/hold

**Comment by gtsiolis** [comment]
Not sure why, but the previous build[[1](https://werft.gitpod-dev.com/job/gitpod-build-se-rename-team.1)] failed. 🤷 

/werft run with-preview

**Comment by gtsiolis** [comment]
Following the hint[[1](https://gitpod.slack.com/archives/C032A46PWR0/p1674746958832439?thread_ts=1674746848.329889&cid=C032A46PWR0)] (internal) how to run this from @AlexTugarev! 💡 

/werft run with-clean-slate-deployment=true recreate-vm=true

**Comment by Siddhant-K-code** [comment]
_(not related to this PR)_: [Cookie name from `team-selection` -> `org-selection` ](https://github.com/gitpod-io/gitpod/blob/3651262ba27bfd270de23c9cb50316007edebc29/components/dashboard/src/teams/teams-context.tsx#L33
)

**Comment by gtsiolis** [comment]
@svenefftinge In case you haven't seen this already, I'm getting this error when trying to login.

Screenshot 2023-01-27 at 11 19 23 AM

**Comment by svenefftinge** [comment]
> Currently, All Projects have a static URL with `<team>` (i.e. `/t/<team>/<project>`). Related [docs](https://www.gitpod.io/docs/configure/projects). This should also be changed because in Docs We need to mention the updated slug URL (`/o/<org>/<project>)`.
> 
> Related Unchanged Files:
> 
> * [/projects/Projects.tsx](https://github.com/gitpod-io/gitpod/blob/e3d8d9151be54a7c705ebb6d176ed0821f816e72/components/dashboard/src/projects/Projects.tsx)
> * [/projects/ProjectListItem.tsx](https://github.com/gitpod-io/gitpod/blob/e3d8d9151be54a7c705ebb6d176ed0821f816e72/components/dashboard/src/projects/ProjectListItem.tsx)

I'll look at those URLs as part of https://github.com/gitpod-io/gitpod/issues/16053

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Review by gtsiolis — COMMENTED** [review]
Thanks, @svenefftinge! 🧙 

Left some UX comments below. 🏀

File: components/dashboard/src/teams/NewTeam.tsx [hunk 1]
```diff
@@ -38,28 +38,28 @@ export default function () {
...
+            <form className="mt-16" onSubmit={createTeam}>
                 <div className="rounded-xl p-6 bg-gray-50 dark:bg-gray-800">
-                    <h3 className="text-left text-lg">You're creating a new team</h3>
-                    <p className="text-gray-500">After creating a team, you can invite others to join.</p>
+                    <h3 className="text-left text-lg">You're creating a new organization</h3>
+                    <p className="text-gray-500">After creating an organization, you can invite others to join.</p>
                     <br />
-                    <h4>Team Name</h4>
+                    <h4>Organization Name</h4>
                     <input
```
[the remark this answers is not in the document]
Comment (gtsiolis) [reply components/dashboard/src/teams/NewTeam.tsx:63]: **issue**: The same issue affects the form on the `/new` flow, see comment below in https://github.com/gitpod-io/gitpod/pull/16050#discussion_r1088875545.

File: components/dashboard/src/teams/NewTeam.tsx [hunk 2]
```diff
@@ -38,15 +38,15 @@ export default function () {
     };
 
     useEffect(() => {
-        document.title = "New Team — Gitpod";
+        document.title = "New Organization — Gitpod";
     }, []);
 
     return (
         <div className="flex flex-col w-96 mt-24 mx-auto items-center">
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/dashboard/src/teams/NewTeam.tsx:45]: does w-112 exist?

[the remark this answers is not in the document]
Comment (gtsiolis) [reply components/dashboard/src/teams/NewTeam.tsx:45]: Ah, sorry! This was from an earlier draft comment. It doesn't, but it works! 🤷

We've also used this once:

https://github.com/gitpod-io/gitpod/blob/605eee48c3f648ed7ec2193339c12e4a7a883291/components/dashboard/src/settings/SSHKeys.tsx#L213

homer-simpson

File: components/dashboard/src/teams/TeamSettings.tsx [hunk 3]
```diff
@@ -157,17 +157,17 @@ export default function TeamSettings() {
...
+                        Update Organization Name
                     </button>
                 </div>
 
-                <h3 className="pt-12">Delete Team</h3>
+                <h3 className="pt-12">Delete Organization</h3>
                 <p className="text-base text-gray-500 pb-4 max-w-2xl">
-                    Deleting this team will also remove all associated data with this team, including projects and
-                    workspaces. Deleted teams cannot be restored!
+                    Deleting this organization will also remove all associated data, including projects and workspaces.
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/dashboard/src/teams/TeamSettings.tsx:166]: We will soon want to also delete the workspaces. So let's just keep this information as it's going to be consistent eventually :-).

File: components/dashboard/src/projects/NewProject.tsx [hunk 4]
```diff
@@ -531,7 +531,7 @@ export default function NewProject() {
         const teamsToRender = teams || [];
         return (
             <>
-                <p className="mt-2 text-gray-500 text-center text-base">Select team or personal account</p>
+                <p className="mt-2 text-gray-500 text-center text-base">Select oranization or personal account</p>
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/dashboard/src/projects/NewProject.tsx:534]: yes, and name it 'organization'

File: components/dashboard/src/teams/TeamSettings.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (gtsiolis) [reply components/dashboard/src/teams/TeamSettings.tsx:166]: agree

**Review by easyCZ — COMMENTED** [review]
There are errors returned from `/workspace/gitpod/components/public-api-server/pkg/apiv1/team.go` which mention teams. These are returned to API calls from the dashboard, and should be changed also

**Comment by easyCZ** [comment]
Spoke sync, we'll follow-up on the public-api changes.

**Review by easyCZ — APPROVED** [review]

**Pull request merged**